amy/dashboard/tests/test_get_involved_views.py

```diff
@@ -0,0 +1,234 @@
+        # check that "trainee_notes" field error IS displayed
+        self.assertContains(
+            rv,
+            'This field is required for activity "Other".',
+            html=True,
+        )
+        # check that "notes" field error IS NOT displayed
+        self.assertNotContains(
+            rv,
+            'This field is required for activity "Other" '
+            "if there are no notes from the trainee.",
+            html=True,
+        )
+        # confirm that no TrainingProgress was created
+        self.assertEqual(len(TrainingProgress.objects.all()), 0)
+
+
+class TestGetInvolvedUpdateView(TestBase):
+    def setUp(self):
+        super()._setUpUsersAndLogin()
+
+        self.get_involved, _ = TrainingRequirement.objects.get_or_create(
+            name="Get Involved",
+            defaults={
+                "url_required": False,
+                "event_required": False,
+                "involvement_required": True,
+            },
+        )
+        self.involvement, _ = Involvement.objects.get_or_create(
+            name="GitHub Contribution",
+            defaults={
+                "display_name": "GitHub Contribution",
+                "url_required": True,
+                "date_required": True,
+            },
+        )
+
+        self.progress = TrainingProgress.objects.create(
+            trainee=self.admin,
+            state="n",
+            requirement=self.get_involved,
+            involvement_type=self.involvement,
+            url="https://example.org",
+            date=date(2023, 7, 27),
+        )
+
+    def test_edit_view_loads(self):
+        # Act
+        rv = self.client.get(reverse("getinvolved_update", args=[self.progress.pk]))
+
+        # Assert
+        self.assertEqual(rv.status_code, 200)
+
+
+class TestGetInvolvedDeleteView(TestBase):
+    def setUp(self):
+        super()._setUpUsersAndLogin()
+
+        self.get_involved, _ = TrainingRequirement.objects.get_or_create(
+            name="Get Involved",
+            defaults={
+                "url_required": False,
+                "event_required": False,
+                "involvement_required": True,
+            },
+        )
+        self.involvement, _ = Involvement.objects.get_or_create(
+            name="GitHub Contribution",
+            defaults={
+                "display_name": "GitHub Contribution",
+                "url_required": True,
+                "date_required": True,
+            },
+        )
+        self.progress = TrainingProgress.objects.create(
+            trainee=self.admin,
+            state="n",
+            requirement=self.get_involved,
+            involvement_type=self.involvement,
+            url="https://example.org",
+            date=date(2023, 7, 27),
+        )
+
+    def test_delete_view_get_request_not_allowed(self):
+        # Act
+        rv = self.client.get(reverse("getinvolved_delete", args=[self.progress.pk]))
+
+        # Assert
+        self.assertEqual(rv.status_code, 405)
```

Comment:
You probably don't have to test this, I hope it has been tested by tests for AMYDeleteView.

Reply:
I can't find any tests for AMYDeleteView in our code - though I would similarly hope this case is covered by tests for the underlying Django DeleteView.
(I copied many of these tests over from `TestCRUDViews` in `test_training_progress.py` - this is one of those)

Reply:
405 is our custom mechanism for handling `GET` requests performed on this delete view. If we don't have tests specific for AMYDeleteView then it's okay to keep this test here.